base/collection_xattr_common.go

```
@@ -275,6 +277,56 @@ func SetXattr(store SubdocXattrStore, k string, xattrKey string, xv []byte) (cas
 
 }
 
+func RemoveXattr(store SubdocXattrStore, k string, xattrKey string, cas uint64) error {
+	worker := func() (shouldRetry bool, err error, value interface{}) {
+		writeErr := store.SubdocDeleteXattr(k, xattrKey, cas)
+		if writeErr == nil {
+			return false, nil, nil
+		}
+
+		if errors.Is(err, gocb.ErrCasMismatch) {
```

The common wrappers in collection_xattr_common shouldn't have any gocb version-specific code, including error handling.  Could use IsCasMismatch here, but since we're just returning the error anyway, I think we can just omit this check and have RemoveXattr return the underlying error, so that callers could check for IsCasMismatch.

If you really think we need to return a typed error here, I think it should be a new ErrCasMismatch error.  ErrCasFailureShouldRetry has specific meaning for the WriteUpdate operations.  And any new error should be added to IsCasMismatch to ensure it continues working as expected.